strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/event/RemoteRepositorySearchEvent.java

```diff
@@ -0,0 +1,22 @@
+package org.carlspring.strongbox.providers.repository.event;
+
+import org.carlspring.strongbox.event.Event;
+import org.carlspring.strongbox.providers.repository.RepositorySearchRequest;
+
+public class RemoteRepositorySearchEvent extends Event
```

Comment:
consider extension of `RepositoryEvent` or `RepositoryBasedEvent` instead of `Event`

Reply:
hm... I don't think that `RepositoryBasedEvent` will suit well because I don't have `path` there.

Reply:
`RepositoryBasedEvent` does not require `path`. It could be helpful in case of searching all repository related events. If you don't agree, skip this comment ;)